src/network/networkd-network-gperf.gperf

```diff
@@ -257,6 +257,8 @@ IPv6AcceptRA.PrefixDenyList,                 config_parse_ndisc_address_filter,
 IPv6AcceptRA.RouteAllowList,                 config_parse_ndisc_address_filter,                        0,                             offsetof(Network, ndisc_allow_listed_route_prefix)
 IPv6AcceptRA.RouteDenyList,                  config_parse_ndisc_address_filter,                        0,                             offsetof(Network, ndisc_deny_listed_route_prefix)
 DHCPServer.RelayTarget,                      config_parse_dhcp_server_relay_target,                    0,                             0
+DHCPServer.RelayAgentCircuitId,              config_parse_string,                                      0,                             offsetof(Network, agent_circuit_id)
+DHCPServer.RelayAgentRemoteId,               config_parse_string,                                      0,                             offsetof(Network, agent_remote_id)
```

Comment:
this should have validating parsers, given that the stuff apparently needs to start with "string:"? why that btw?

Reply:
Again this discussions explains the rationale https://github.com/systemd/systemd/pull/19384#discussion_r620488412

Reply:
made a specialized parser function. Ready for another review